OAC csv parsing error

The cause of this issue is the HTML tag in the `alias` filed, that doesn't parse properly when using the csv data format.

Suggested workaround: remote data_format kwarg as it shouldn't matter for the end user what server output format we parse internally.

Cross-reference: https://github.com/astrocatalogs/OACAPI/issues/11

==============

There are a few new(ish) test failures with OAC: https://travis-ci.org/astropy/astroquery/jobs/512564435#L4631

```
______________________ TestOACClass.test_query_object_csv ______________________
self = <astroquery.oac.tests.test_oac_remote.TestOACClass object at 0x7f8ef4fdfb00>
    def test_query_object_csv(self):
>       phot = OAC.query_object(event='SN2014J')
astroquery/oac/tests/test_oac_remote.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astroquery/utils/class_or_instance.py:25: in f
    return self.fn(obj, *args, **kwds)
astroquery/utils/process_asyncs.py:29: in newmethod
    result = self._parse_result(response, verbose=verbose)
astroquery/oac/core.py:487: in _parse_result
    output_response = self._format_output(raw_output)
astroquery/oac/core.py:451: in _format_output
    output = json.loads(raw_output)
/home/travis/miniconda/envs/test/lib/python3.7/json/__init__.py:348: in loads
    return _default_decoder.decode(s)
/home/travis/miniconda/envs/test/lib/python3.7/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <json.decoder.JSONDecoder object at 0x7f8f03188fd0>
s = 'event,escapevelocity,stellarclass,lumdist,ra,references,maxappmag,hostdec,host,instruments,sources,redshift,hostra,ma...y et al.,Steve J. Fossey",,2014/01/21,300,,,"+69:40:25.9,+69:40:26.00,+69:40:26.0",,e,-18.4,,0.1358,SN2014J,"Ia,Ia-HV"'
idx = 0
    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
/home/travis/miniconda/envs/test/lib/python3.7/json/decoder.py:355: JSONDecodeError
----------------------------- Captured stdout call -----------------------------
The API did not return a valid CSV output! 
Outputing JSON-compliant dictionary instead.
____________________ TestOACClass.test_query_region_box_csv ____________________
self = <astroquery.oac.tests.test_oac_remote.TestOACClass object at 0x7f8ef400cb38>
    def test_query_region_box_csv(self):
        phot = OAC.query_region(coordinates=self.test_coords,
                                width=self.test_width,
>                               height=self.test_height)
astroquery/oac/tests/test_oac_remote.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astroquery/utils/class_or_instance.py:25: in f
    return self.fn(obj, *args, **kwds)
astroquery/utils/process_asyncs.py:29: in newmethod
    result = self._parse_result(response, verbose=verbose)
astroquery/oac/core.py:487: in _parse_result
    output_response = self._format_output(raw_output)
astroquery/oac/core.py:451: in _format_output
    output = json.loads(raw_output)
/home/travis/miniconda/envs/test/lib/python3.7/json/__init__.py:348: in loads
    return _default_decoder.decode(s)
/home/travis/miniconda/envs/test/lib/python3.7/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <json.decoder.JSONDecoder object at 0x7f8f03188fd0>
s = 'event,escapevelocity,stellarclass,lumdist,ra,references,maxappmag,hostdec,host,instruments,sources,redshift,hostra,ma...y et al.,Steve J. Fossey",,2014/01/21,300,,,"+69:40:25.9,+69:40:26.00,+69:40:26.0",,e,-18.4,,0.1358,SN2014J,"Ia,Ia-HV"'
idx = 0
    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
/home/travis/miniconda/envs/test/lib/python3.7/json/decoder.py:355: JSONDecodeError
----------------------------- Captured stdout call -----------------------------
The API did not return a valid CSV output! 
Outputing JSON-compliant dictionary instead.
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -38,6 +38,13 @@ linelists.cdms
 - Fix issues with the line name parser and the line data parser; the original
   implementation was incomplete and upstream was not fully documented. [#2385, #2411]
 
+oac
+^^^
+
+- Fix bug in parsing events that contain html tags (e.g. in their alias
+  field). [#2423]
+
+
 Infrastructure, Utility and Other Changes and Additions
 -------------------------------------------------------
 

--- a/astroquery/oac/core.py
+++ b/astroquery/oac/core.py
@@ -12,6 +12,7 @@ and James Guillochon (jguillochon@cfa.harvard.edu)
 
 import json
 import csv
+import re
 
 import astropy.units as u
 from astropy.table import Column, Table
@@ -435,7 +436,12 @@ class OACClass(BaseQuery):
 
     def _format_output(self, raw_output):
         if self.FORMAT == 'csv':
-            split_output = raw_output.splitlines()
+
+            fixed_raw_output = re.sub('<[^<]+?>', '', raw_output)
+            split_output = fixed_raw_output.splitlines()
+
+            # Remove any HTML tags
+
             columns = list(csv.reader([split_output[0]], delimiter=',',
                            quotechar='"'))[0]
             rows = split_output[1:]

--- a/astroquery/oac/tests/test_oac_remote.py
+++ b/astroquery/oac/tests/test_oac_remote.py
@@ -59,25 +59,27 @@ class TestOACClass:
                                 data_format='json')
         assert isinstance(phot, dict)
 
-    @pytest.mark.xfail(reason="Upstream API issue.  See #1130")
     def test_get_photometry(self):
         phot = OAC.get_photometry(event="SN2014J")
         assert isinstance(phot, Table)
+        assert len(phot) > 0
 
     def test_get_photometry_b(self):
         phot = OAC.get_photometry(event="SN2014J")
         assert isinstance(phot, Table)
+        assert len(phot) > 0
 
-    @pytest.mark.xfail(reason="Upstream API issue.  See #1130")
     def test_get_single_spectrum(self):
         spec = OAC.get_single_spectrum(event="SN2014J",
                                        time=self.test_time)
         assert isinstance(spec, Table)
+        assert len(spec) > 0
 
     def test_get_single_spectrum_b(self):
         test_time = 56680
         spec = OAC.get_single_spectrum(event="SN2014J", time=test_time)
         assert isinstance(spec, Table)
+        assert len(spec) > 0
 
     def test_get_spectra(self):
         spec = OAC.get_spectra(event="SN2014J")

